packages/pigeon/lib/objc_generator.dart

```diff
@@ -62,48 +62,64 @@ String _className(String? prefix, String className) {
   }
 }
 
-String _callbackForType(TypeDeclaration type, String objcType) {
+String _callbackForType(TypeDeclaration type, _ObjcPtr objcType) {
   return type.isVoid
       ? 'void(^)(NSError *_Nullable)'
-      : 'void(^)($objcType *, NSError *_Nullable)';
+      : 'void(^)(${objcType.ptr}, NSError *_Nullable)';
 }
 
-const Map<String, String> _objcTypeForDartTypeMap = <String, String>{
-  'bool': 'NSNumber',
-  'int': 'NSNumber',
-  'String': 'NSString',
-  'double': 'NSNumber',
-  'Uint8List': 'FlutterStandardTypedData',
-  'Int32List': 'FlutterStandardTypedData',
-  'Int64List': 'FlutterStandardTypedData',
-  'Float64List': 'FlutterStandardTypedData',
-  'List': 'NSArray',
-  'Map': 'NSDictionary',
+class _ObjcPtr {
+  const _ObjcPtr({required this.baseName, required this.hasAsterisk});
+  final String baseName;
+  final bool hasAsterisk;
```

Comment:
If it's representing a pointer, this can only be true if `baseName` is `id`, right? In which case, this could be a getter rather than a constructor property.